urllib3/util/retry.py

```
@@ -194,12 +206,51 @@ def get_backoff_time(self):
         backoff_value = self.backoff_factor * (2 ** (consecutive_errors_len - 1))
         return min(self.BACKOFF_MAX, backoff_value)
 
-    def sleep(self):
-        """ Sleep between retry attempts using an exponential backoff.
+    def parse_retry_after(self, retry_after):
+        if retry_after.isdigit():
+            seconds = int(retry_after)
+        else:
+            retry_date_tuple = email.utils.parsedate(retry_after)
+            if retry_date_tuple is None:
+                raise ValueError("Invalid Retry-After header: %s" % retry_after)
```

Can we raise a urllib3 exception here? `ValueError` is going to be unexpected for a lot of users.